src/common/sentry.ts

```diff
@@ -21,95 +21,78 @@
+const environment = isProduction ? "production" : "development";
 
-const logInitError = (reason: string) => {
+function logInitError(reason: string) {
   logger.warn(`⚠️  [SENTRY-INIT]: ${reason}, Sentry is not initialized.`);
-};
-
-export const beforeSend = (event: SentryEvent) => {
-  const allowErrorReporting = UserStore.getInstance().allowErrorReporting;
+}
 
-  if (allowErrorReporting) return event;
+function beforeSend(event: SentryEvent) {
+  if (UserStore.getInstance().allowErrorReporting) {
+    return event;
+  }
 
-  logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: ${allowErrorReporting}. Sentry event is caught but not sent to server.`);
+  logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: false. Sentry event is caught but not sent to server.`);
   logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: === START OF SENTRY EVENT ===");
   logger.info(event);
   logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: ===  END OF SENTRY EVENT  ===");
 
   // if return null, the event won't be sent
   // ref https://github.com/getsentry/sentry-javascript/issues/2039
   return null;
-};
-
-export const SentryInit = () => {
-  if (!sentryDsnIsValid) {
-    logInitError("Sentry DSN is not valid");
+}
 
-    return;
+async  function requireSentry() {
+  if (processType === "browser") {
+    return import("@sentry/electron/dist/main");
   }
 
-  const processType = process.type as typeof electronProcessType[number];
-
-  let Sentry;
-
-  try {
-    // if in main process 
-    if (processType === "browser") {
-      Sentry = require("@sentry/electron/dist/main");
-    }
-
-    // if in renderer process
-    if (processType === "renderer") {
-      Sentry = require("@sentry/electron/dist/renderer");
-    }
-  } catch (error) {
-    logInitError(`Error loading Sentry module ${error?.message}`);
-
-    return;
+  if (processType === "renderer") {
+    return import("@sentry/electron/dist/renderer");
   }
 
+  throw new Error(`Unsupported process type ${processType}`);
+}
 
-  if (!Sentry) {
-    logInitError(`Unsupported process type ${processType}`);
-
-    return;
-  }
-
+export async function SentryInit() {
   try {
-    Sentry.init({
-      beforeSend,
-      dsn: sentryDsn,
-      integrations,
-      initialScope: initialScope(processType),
-      environment
-    });
-    logger.info(`✔️  [SENTRY-INIT]: Sentry for ${processType} is initialized.`);
+    const Sentry = await requireSentry();
+
+    try {
+      Sentry.init({
+        beforeSend,
+        dsn: sentryDsn,
+        integrations,
+        initialScope,
+        environment
+      });
+      logger.info(`✔️  [SENTRY-INIT]: Sentry for ${processType} is initialized.`);
+    } catch (error) {
+      return logInitError(`Sentry.init() error ${error?.message}`);
+    }
   } catch (error) {
-    logInitError(`Sentry.init() error ${error?.message}`);
-
-    return;
+    return logInitError(`Error loading Sentry module ${error?.message ?? error}`);
```

Comment:
I wouldn't return `logInitError()` here, might not matter here because `logInitError` also doesn't return, but I think it's better to return just what intended to return in the case of early return. (For example, if `logInitError()` returns `string`, `SentryInit` will be void or string, most linter doesn't like that..

Reply:
I reverted this.